<launch>
  <!-- <param name="/use_sim_time" value="true"/> -->
  <include file="$(find frontier_exploration)/launch/stage.launch"/>
  <node name="rviz" pkg="rviz" type="rviz" required="true" args="-d $(find state_estimator)/config/lab2.rviz" >
    <remap from="/move_base_simple/goal" to="/mce234b/goal_pose_published"/>
  </node>
        
  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 odom map 100" />
    <!-- Run the map and global planner node -->
  <node name="map_sim" pkg="frontier_exploration" type="sim_map_pub.py" output="screen" cwd="node"/>
  <!-- Publishes the voxel grid to rviz for display -->
  <!-- Run the costmap node -->
  <!-- <node name="costmap_node" pkg="costmap_2d" type="costmap_2d_node" >
    <rosparam file="$(find frontier_exploration)/move_base_config/costmap_standalone.yaml" command="load" ns="costmap" />
  </node> -->
  <!-- Run the global planner node -->
  <node name="global_planner" pkg="global_planner" type="planner" respawn="false" output="screen">
      <rosparam file="$(find frontier_exploration)/move_base_config/global_planner.yaml" command="load" />
      <rosparam file="$(find frontier_exploration)/move_base_config/costmap_standalone.yaml" command="load" ns="costmap" />

    </node>
  
  <node name="navigation" pkg="frontier_exploration" type="prelab3b2_nav.py" output="screen" cwd="node"/>

  <!-- <node name="global" pkg="frontier_exploration" type="prelab3b2_plan.py" output="screen" cwd="node"/> -->
  <!-- Run the local planner node -->
  <node name="local" pkg="frontier_exploration" type="mpc_local_planner.py" output="screen" cwd="node">
    <remap from="/cmd_vel/teleopDrive" to="/cmd_vel"/>
  </node>

    <!-- <rosparam file="$(find frontier_exploration)/move_base_config/global_planner.yaml" command="load" /> -->
</launch>